docs: add initial experimental vitest testing guide (#61483)
angular/angular: adev/src/content/guide/testing/experimental-unit-test.md

# Experimental unit testing system

> The Angular CLI also provides an experimental unit test system that can use [Vitest](https://vitest.dev/) as a test runner.

Reviewer: ```suggestion
The Angular CLI provides an experimental unit test system that can use [Vitest](https://vitest.dev/) as a test runner.
```

IMPORTANT: This experimental unit testing system requires the use of the `application` build system.
This build system is the default for all newly created projects.

## Set up testing

The Angular CLI includes the test system within a new project but must be configured before it can be used.

The project you create with the CLI is setup to use the `karma` test system by default.

> To change to the experimental unit test system, the `test` target should be updated:

Reviewer: ```suggestion
To change to the experimental unit test system, update the `test` target as follows:
```

passive => active voice

<docs-code language="json">
"test": {
  "builder": "@angular/build:unit-test",
  "options": {
    "tsConfig": "tsconfig.spec.json",
    "runner": "vitest",
    "buildTarget": "::development",
  }
}
</docs-code>

The `buildTarget` operates similarly to the option available to the development server.

> Build options for the tests are configured within the `build` target itself.

Reviewer: ```suggestion
The `build` target configures build options for the tests.
```

passive => active voice

> If the `development` build configuration does not exist for a project or different options are needed for testing, a `testing` or similarly named build configuration can be created and used instead.

Reviewer: ```suggestion
If the `development` build configuration is missing for a project or you need
different options for testing, you can create and use a `testing` or similarly named build configuration.
```

* negative (does not exist) => positive construction (is missing)
* passive => active voice
